### PR TITLE
Addition of filterProperty on RuleInfo to support filtering on code

### PR DESCRIPTION
**Pull request by pandurangpatil**
Addition of filterProperty on RuleInfo to support filtering on code property of the node along with methodFullName.

1. Made changes in the code to add a new property `filterProperty` on RuleInfo. Along with changes to set the default value as `MethodFullName`
2. Made respective changes where RuleInfo object has been instantiated in the code to set the default value as `MethodFullName`
3. Sink tagger changes for JS Regular sink tagger and generic sink tagger
4. Unit tests along with rule validator changes.

### CODE REVIEW

File: src/main/resources/ai/privado/rulevalidator/schema/sinks.json [hunk 1]
```diff
@@ -45,6 +45,17 @@
...
             "pattern": "^.*$"
           },
+          "filterProperty": {
+            "$id": "#root/sinks/items/filterProperty",
+            "title": "FilterProperty",
+            "type": "string",
+            "default": "method_full_name",
+            "examples": [
+              "code",
+              "method_full_name"
```
Comment (tuxology) [inline src/main/resources/ai/privado/rulevalidator/schema/sinks.json:55]: nitpick: should we use `camelCase` key for `methodFullName`?

**Review by tuxology — DISMISSED** [review]
LGTM 🌈  Only small nitpick

File: src/main/resources/ai/privado/rulevalidator/schema/sinks.json (hunk 1 above)
Comment (pandurangpatil) [reply src/main/resources/ai/privado/rulevalidator/schema/sinks.json:55]: We have followed underscore-separated values for values like category = third_parties. So I continued with a similar pattern.

Comment (tuxology) [reply src/main/resources/ai/privado/rulevalidator/schema/sinks.json:55]: Ok, makes sense! Lets merge once tests are done 👍

**Review by tuxology — APPROVED** [review]

**Pull request merged**